cni/pkg/nodeagent/server.go

```diff
@@ -66,6 +68,46 @@ type Server struct {
 	cniServerStopFunc func()
 }
 
+// runWithHostNs executes the provided function within the host network namespace
+func runWithHostNs(f func() error) error {
+	if f == nil {
+		return nil
+	}
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	currentNS, err := netns.GetCurrentNS()
```

Comment:
It is probably better if use netNs.WithNetNSPath/Do instead of re-implementing the switch logic